Edit persona, multiple fields, and avatar

jessgusclark opened the pull request:
Features:

- Add avatar based on identicon using the DID.
- Modal to edit persona items, use existing DataVaultKey type to keep it consistent.
- Persona items are prefixed with `DD_`, however, the user can add their own DD items in the DataVault screen. This may cause issues if a user creates a new item, i.e. `DD_NAME`. On refresh, the Persona will take the "first" item returned.
- Edit multiple fields at once, script decides if the item should be ignored(same value), inserted, edited, or removed.
- Always return DIDs with `:ethr:`, and removes variable to make this optional.

ilanolkies commented on src/formatters.ts:
```diff
@@ -7,8 +7,8 @@ export const displayIdentity = (address: string, chainId?: number) => {
-    case 30: return `did:rsk:${addr}`
-    case 31: return `did:rsk:testnet:${addr}`
+    case 30: return `did:ethr:rsk:${addr}`
+    case 31: return `did:ethr:rsk:testnet:${addr}`
```
Why don't using `createDidFormat`?

jessgusclark replied:
hahahah! we are!!! In the next PR you are looking at, this function is going away completely!! I didn't remove it in this PR since I wasn't touching the code that interacted with it.

ilanolkies replied:
🙌

ilanolkies commented on src/app/state/operations/datavault.ts:
```diff
@@ -78,3 +78,31 @@ export const swapDataVaultContent = (client: DataVaultWebClient, key: string, co
 export const getStorageInformation = (client: DataVaultWebClient) => (dispatch: Dispatch<any>) =>
   client.getStorageInformation()
     .then((storage: DataVaultStorageState) => dispatch(receiveStorageInformation({ storage })))
+    .catch((err: Error) => console.log('could not get storage', err))
```
This hides the error

jessgusclark replied:
I will remove that line to allow the error to show.

ilanolkies reviewed (changes requested):
LGTM

When I hover my name I see my mouse with a question mark. Why is this?

jessgusclark commented:
> When I hover my name I see my mouse with a question mark. Why is this?

This is a mistake, the hover state shouldn't exist if there is a name there. I'll update this to show only the name if it exists.

jessgusclark commented:
Rebased after merging other PR. The formatter was removed in the previous PR, and removed the catch statement for the DV. 

As of writing this, the DV is down so this is not testable right now (https://identity.staging.rifcomputing.net/__health)
